x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/join/LookupJoin.java

```diff
@@ -71,4 +75,14 @@ protected NodeInfo<Join> info() {
             config().rightFields()
         );
     }
+
+    @Override
+    public void postAnalysisVerification(Failures failures) {
```

Comment:
Nit: While I think using `PostAnalysisVerificationAware` is preferable, now the checks for lookup join are split between here and directly in `Analyzer.resolveIndex`: https://github.com/elastic/elasticsearch/blob/e2f2ac35c2f91f101dc049a3f06a184c78c69873/x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java#L240-L256

If it's not too much work, could we consolidate the two?

Reply:
I guess the check in `Analyzer` is generic (ie. doesn't refer to LOOKUP) only because it's easier and because we use IndexMode.LOOKUP only in LOOKUP JOIN.
I'll move it to LookupJoin.

Reply:
Yeah. I think it's actually clearer the way you moved it.

This relied on the fact that we assign `IndexMode.LOOKUP` at _parsing time_ to the `UnresolvedRelation` obtained from parsing `LOOKUP JOIN`. But that's more convoluted than your PR's current solution.